### PR TITLE
Memberships API

### PR DESCRIPTION
**Pull request by iamdevice**

### CODE REVIEW

File: spec/components/schemas/Merchant.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  name:
+    description: The merchant name
+    type: string
+  email:
+    description: The merchant email
+    type: string
+    format: email
+    maxLength: 100
+  lastName:
+    description: The merchang user's last name
```
Comment (wmdmgv) [inline spec/components/schemas/Merchant.yaml]: typo merchang

File: spec/components/schemas/Merchant.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    type: string
+  firstName:
+    description: The merchant user's first name
+    type: string
+  reportCurrency:
+    description: The user's ISO Alpha-3 code used for reports
+    readOnly: true
+    type: string
+  status:
+    description: The user status
```
Comment (wmdmgv) [inline spec/components/schemas/Merchant.yaml]: Maybe Merchant status? also, seems we have **active** and **inactive** status for merchants only.

File: spec/paths/memberships.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    - Memberships
+  summary: Create or update membership
+  operationId: PutMembership
+  description: |
+    Create or update membership using merchant and user identifiers
+  requestBody:
+    $ref: '#/components/requestBodies/Membership'
+  responses:
+    201:
+      description: Membershop was created
```
Comment (wmdmgv) [inline spec/paths/memberships.yaml]: typo MembershOp ?

File: spec/components/schemas/Membership.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    allOf:
+      - $ref: "#/components/schemas/Merchant"
+  user:
+    description: Membership user
+    allOf:
+      - $ref: "#/components/schemas/User"
+  isOwner:
+    description: User is owner of merchant
+    type: boolean
+    readOnly: true
```
Comment (wmdmgv) [inline spec/components/schemas/Membership.yaml]: @king555123 Should we able to change isOwner and isDefault by API?

File: spec/components/requestBodies/Membership.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+content:
+  application/json:
+    schema:
+      properties:
+        merchantId:
```
Comment (slavcodev) [inline spec/components/requestBodies/Membership.yaml]: https://github.com/Rebilly/RebillyUserAPI/pull/291#discussion_r325858900

File: spec/components/schemas/Membership.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+type: object
+required:
+  - merchant
+  - user
+properties:
+  merchant:
```
Comment (slavcodev) [inline spec/components/schemas/Membership.yaml]: https://github.com/Rebilly/RebillyUserAPI/pull/291#discussion_r325858900

**Comment by iamdevice** [comment]
devs please re-review

File: spec/components/schemas/Membership.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+          - $ref: '#/components/schemas/ResourceId'
+      name:
+        description: First and last name
+        type: string
+  permissions:
+    description: The user's permissions. See the format in example
+    allOf:
+      - $ref: "#/components/schemas/AclPermissions"
+  isOwner:
+    description: User is owner of merchant
```
Comment (slavcodev) [inline spec/components/schemas/Membership.yaml]:
```suggestion
    description: User is owner of organization
```

File: spec/components/schemas/Membership.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    type: boolean
+  isDefault:
+    description: This merchant is default for membership user
```
Comment (slavcodev) [inline spec/components/schemas/Membership.yaml]:
```suggestion
    description: This organization is default for membership user
```

File: spec/openapi.yaml [hunk 9]
```diff
@@ -209,6 +209,10 @@ tags:
       Segments are grid UI settings that can be shared with other users. They are like a custom view.
       For example, one may set up a transaction grid to show specific columns and sort a certain way, and share it with
       others in their organization.
+  - name: Memberships
+    description: |
+      Membership describe the relationships between merchants and users.
```
Comment (slavcodev) [inline spec/openapi.yaml]:
```suggestion
      Membership describe the relationships between organinizations and users.
```

File: spec/paths/memberships@{organizationId}@{userId}.yaml [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+    description: User identifier
+    schema:
+      $ref: '#/components/parameters/resourceId'
+get:
+  tags:
+    - Memberships
+  summary: Retrieve the membership
+  operationId: GetMembership
+  description: |
+    Retrieve the membership using merchant and user identifiers
```
Comment (slavcodev) [inline spec/paths/memberships@{organizationId}@{userId}.yaml]:
```suggestion
    Retrieve the membership using organization and user identifiers
```

File: spec/openapi.yaml [hunk 11]
```diff
@@ -209,6 +209,10 @@ tags:
...
+      For example, one user could have more than one merchants.
```
Comment (slavcodev) [inline spec/openapi.yaml]:
```suggestion
      For example, one user could have more than one organizations.
```

**Pull request merged**